api/src/test/java/org/openmrs/module/fhir2/api/dao/impl/FhirObservationDaoImplTest.java

```diff
@@ -477,21 +478,94 @@ public void searchForObs_shouldSearchForObsByValueDate() {
 	}
 	
 	@Test
-	public void searchForObs_shouldSearchForObsByValueQuantity() {
+	public void searchForObs_shouldSearchForObsByValueQuantityWithPrefixAp() {
 		QuantityAndListParam quantityAndListParam = new QuantityAndListParam();
 		
 		QuantityOrListParam quantityOrListParam = new QuantityOrListParam();
 		
 		QuantityParam quantityParam = new QuantityParam();
-		quantityParam.setValue(VALUE_QUANTITY);
+		quantityParam.setValue("36");
+		quantityParam.setPrefix(ParamPrefixEnum.APPROXIMATE);
 		quantityAndListParam.addAnd(quantityOrListParam.add(quantityParam));
 		
 		Collection<Obs> results = dao.searchForObservations(null, null, null, null, null, quantityAndListParam, null, null,
 		    null, null);
 		
 		assertThat(results, notNullValue());
 		assertThat(results, not(empty()));
-		assertThat(results, hasItem(hasProperty("uuid", equalTo(OBS_VALUE_CONCEPT_UUID))));
+		assertThat(results, hasItem(hasProperty("uuid", equalTo("7e77d071-1f7d-4394-a316-0a458edf28c3"))));
+	}
+	
+	@Test
+	public void searchForObs_shouldSearchForObsByValueQuantityWithoutPrefixAndDecimalValue() {
+		QuantityAndListParam quantityAndListParam = new QuantityAndListParam();
+		
+		QuantityOrListParam quantityOrListParam = new QuantityOrListParam();
+		
+		QuantityParam quantityParam = new QuantityParam();
+		quantityParam.setValue("100.00");
+		quantityAndListParam.addAnd(quantityOrListParam.add(quantityParam));
+		
+		Collection<Obs> results = dao.searchForObservations(null, null, null, null, null, quantityAndListParam, null, null,
+		    null, null);
+		
+		assertThat(results, notNullValue());
+		assertThat(results, not(empty()));
+		assertThat(results, hasItem(hasProperty("uuid", equalTo("89fg071-1f7d-4394-a316-0a458edf28c3"))));
+	}
+	
+	//	@Test
+	//	public void searchForObs_shouldSearchForObsByValueQuantityWithoutPrefixAndEValue() {
+	//		QuantityAndListParam quantityAndListParam = new QuantityAndListParam();
+	//
+	//		QuantityOrListParam quantityOrListParam = new QuantityOrListParam();
+	//
+	//		QuantityParam quantityParam = new QuantityParam();
+	//		quantityParam.setValue("1e2");
+	//		quantityAndListParam.addAnd(quantityOrListParam.add(quantityParam));
+	//
+	//		Collection<Obs> results = dao.searchForObservations(null, null, null, null, null, quantityAndListParam, null, null,
+	//		    null, null);
+	//
+	//		assertThat(results, notNullValue());
+	//		assertThat(results, not(empty()));
+	//		assertThat(results, hasItem(hasProperty("uuid", equalTo("4efa62d2-6b8b-4803-a8fa-3f32ee54db4f"))));
+	//	}
+	//
+	//	@Test
+	//	public void searchForObs_shouldSearchForObsByValueQuantityWithoutPrefixAndNegativeEValue() {
+	//		QuantityAndListParam quantityAndListParam = new QuantityAndListParam();
+	//
```

Comment:
@ibacher So I have added some tests which don't work at the moment because the prefix coding is not correct for `1e2` and `1e-2` type of numbers.

Reply:
So, for prefix coding, I'm happy with `eq` being an exact equality, e.g. `1e2` means only exactly `100` (despite the spec, this is the actual implementation in the HAPI JPA Server). `ap` should still work, since `1e2` should still translate to `[90 ... 100]`. `le`, `ge`, `lt`, and `gt` are all defined as starting from the exact position at infinite precision, so again should be covered by just the normal Hibernate operations.

Reply:
@ibacher I have pushed a commit after this please have a look and tell if anything needs to be fixed